src/OfficeHours.ts

```diff
@@ -0,0 +1,112 @@
+import { SharedSettings } from "./SharedSettings";
+import { fileBackedObject } from "./FileBackedObject";
+
+import Discord = require("discord.js");
+
+interface QuestionData {
+    uuid: string;
+    author: string;
+    question: string;
+}
+
+export default class OfficeHours {
+    private bot: Discord.Client;
+    private questions: QuestionData[];
+    private adminRoles: string[];
+
+    private openMessage: string;
+    private closeMessage: string;
+    private addedMessage: string;
+    private removedMessage: string;
+
+    constructor(bot: Discord.Client, sharedSettings: SharedSettings, questionFile: string) {
+        console.log("Requested OfficeHours extension..");
+        this.bot = bot;
+
+        this.questions = fileBackedObject(questionFile);
+        console.log("Successfully question file.");
+
+        this.openMessage = sharedSettings.officehours.openMessage;
+        this.closeMessage = sharedSettings.officehours.closeMessage;
+        this.addedMessage = sharedSettings.officehours.addedMessage;
+        this.removedMessage = sharedSettings.officehours.removedMessage;
+        this.adminRoles = sharedSettings.officehours.allowedRoles;
+
```

Comment:
Do we need to assign all of these to a variable? We only use em once, so we might as well inline the references directly.